src/Adapter/Product/SpecificPrice/Repository/SpecificPriceRepository.php

```diff
@@ -125,6 +126,21 @@ public function get(SpecificPriceId $specificPriceId): SpecificPrice
         return $specificPrice;
     }
 
+    /**
+     * @param SpecificPriceId $specificPriceId
+     * @return void
+     */
+    public function delete(SpecificPriceId $specificPriceId): void
+    {
+        $objectModel = $this->getObjectModel(
+            $specificPriceId->getValue(),
+            SpecificPrice::class,
+            SpecificPriceNotFoundException::class
+        );
+
+        $this->deleteObjectModel($objectModel, CannotDeleteSpecificPriceException::class);
```

Comment:
This can be shortedned to:

```suggestion
        $this->deleteObjectModel($this->get($specificPriceId), CannotDeleteSpecificPriceException::class);
```